pyo3-build-config/src/impl_.rs

```diff
@@ -461,6 +484,30 @@ print("mingw", get_platform().startswith("mingw"))
         }
         Ok(())
     }
+
+    /// Run a python script using the executable of this InterpreterConfig with additional
+    /// environment variables (e.g. PYTHONPATH) set.
+    pub fn run_python_script_with_envs<I, K, V>(&self, script: &str, envs: I) -> Result<String>
```

Comment:
Probably worth adding a CHANGELOG entry for this and the below new function.

Would also be great to document they panic if this config doesn't know the interpreter in a `# Panics` section.